### PR TITLE
Created a copy of input tensor for evaluation

### PR DESCRIPTION
**Pull request by vkovinicTT**
### Summary

This PR ensures that a copy of the input tensors is created when running forward during compilation (e.g. when trying to verify outputs using forward pass of the framework model). This prevents unintended modifications to the original input tensors if in-place operations are performed during the forward pass.

### Changes

- Added logic to create copies of input tensors before passing them to the model.
- Ensured that this applies to supported tensor types across different frameworks.
- Prevents unexpected side effects when the model modifies inputs in-place.

### Why is this needed?
Some models perform in-place operations on input tensors during the forward pass, which can lead to unintended changes in the original inputs. By making a copy of the inputs, we ensure correctness and avoid potential issues when running forward during compilation.

Example test:
```python
    class Inplace(torch.nn.Module):
        def __init__(self):
            super().__init__()

        def forward(self, x):
            y = x + 1 
            x += 2 # in-place operation on the input that causes the inputs to change during forge.compile(...)

            return x + y
        
    input = torch.zeros(shape, requires_grad=False)
    framework_input = input.detach().clone()
    tt_inputs = [input]

    framework_model = Inplace()
    y = framework_model(framework_input)
        
    compiled_model = forge.compile(framework_model, sample_inputs=tt_inputs, module_name="inplace")
    tty = compiled_model(*tt_inputs)[0]

    compare_with_golden(golden=y, calculated=tty) # this would fail

### CODE REVIEW

**Review by nvukobratTT — APPROVED** [review]

**Comment by vkovinicTT** [comment]
I was thinking of maybe trying out [functionalize](https://pytorch.org/functorch/nightly/generated/functorch.functionalize.html) for torch as Knezevic has pointed that they do on `torch` frontend. Because as I've seen other frameworks (jax and I think tensorflow) are functional meaning no side effects and no changing of the inputs.
